runtime/caml/domain.h

```diff
@@ -97,6 +97,15 @@ Caml_inline intnat caml_domain_alone(void)
   return atomic_load_acquire(&caml_num_domains_running) == 1;
 }
 
+/* The index of the current domain. It is an integer unique among
+   currently-running domains, in the interval [0; N-1] where N is the
+   peak number of domains running simultaneously so far. The index of
+   a terminated domain may be reused for a new domain. */
+Caml_inline int caml_domain_index(void)
+{
+  return Caml_state->id;
+}
```

Comment:
This function is not mentioned in the changelog. If it is meant to be public, shouldn't it be documented in the manual?

Reply:
This is an excellent question. I looked at it but ended up doing nothing for the following reasons:
- as far as I know, there is no documentation in the manual for the C interface to the domain machinery (there is a chapter on multicore programming, that is pure OCaml; and there is a chapter on the C FFI that mostly does not mention domains)
- this function is in CAML_INTERNALS, as the rest of domain.h, so technically users are not supposed to rely on it

In this context one can wonder whether we should bother exposing this as a function at all -- especially since the definition is trivial. I think that it makes sense to have as an abstraction, for the same reasons that it is useful in Domain, and I could see people writing low-level C extensions for concurrent programming (using CAML_INTERNALS) using it in a way that would be clearer and more forward-compatible as relying on `Caml_state->id` directly.

Reply:
OK, fine by me!